File: Vishwas RNO 40/DotNet/Assignment 1/DemoApp/MathUtil/Assinment/MathUtil.cs
namespace Math;

class MathUtil
{
    public static bool IsEven(int num)
    {
        if(num%2 ==0)
        {
            return true;
        }
        return false;
    }
    
public static bool IsOdd(int num)
    {
    if(num%2 != 0)
    {
        return true;
    }
    return false;
    }

public static bool IsPrime(int num)
    {
    if(num == 1)
        return false;
    if(num == 2 || num == 3)
    return true;

    if((num % 2) == 0)
    return false;
    
    for(int m=3; m * m <= num; m +=2)
    {
        if((num % m) == 0)

        return false;
    }
    return true;
    }

    public static int CountPrimes(int num1, int num2)
    {
        int term;
        int count = 0;
        for(term = num1; term<= num2; ++term)
        {
            if(IsPrime(term))
            {
                ++count;
            }
        }
        return count;
    }

    public static int Reverse(int num)
    {
        int rem = 0;
        int reverse = 0;
        while(num != 0)
    
     {
         rem = num % 10;
         reverse = reverse * 10 + rem;
         num = num / 10;
     
     }
     return reverse;
    }

    public static int DigitCount(int num)
    {
        int count = 0;
        while(num != 0)
        {
            num /=10;
             ++count;
        }
        return count;
    }
}
    

File: java Assignment/Assignment2/MathUtil.java
class MathUtil {
	private int first;
	private int second;
	
	public MathUtil(int a, int b){
		first=a;
		second=b;
	}
	public boolean isEven(int first) {
		boolean flag = false;

		if(first %2==0)
		{
			flag=true;
		}
		return flag;

	}
	public static boolean isOdd(int first) {
		boolean flag = false;
		if(first %2==1)
		{
			flag=true;
		}
		return flag;
	}  

	public boolean isPrime(int first){
		int num = first;
		boolean flag = false;

		for (int i =2; i <=num / 2; ++i){
			if(num % i == 0){
				flag = true;
			}
		}
		return flag;
	}
	
	  public int isprime(int n) {
		int m;
		if(n == 1)
			return 0;
		if(n == 2 || n == 3)
			return 1;
		if((n % 2)== 0)
			return 0;
		for(m = 3; m*m <=n; m+= 2)
		{
			if((n % m) ==0)
			return 0;
		}
		return 1;
				
	}
	
	public  int countPrimes(int first,int second) {
		int term, count=0;
		
        /*        if(first==2)
			return 1;
		if(first==3)
			return 1;*/
		for(term = first; term <= second;++term)
		{
			count += isprime(term);
			
		}
		return count;

	}
	
	public  int reverse(int first) {
                int sum=0;
		int temp=first;
		while(first>0)
		{
			int r=0;
			r = first%10;
			sum=sum*10+r;
			first=first/10;
		}
		return sum;

	}
	
	public int digitCount(int first) {
                int count=0;
		while(first>0)
		{
			int r=0;
			r=first%10;
			count++;
			first=first/10;
		}
		return count;

	}
}	
	
		
